### DISCUSSION TITLE
Teleport applications and reverse proxy

### DISCUSSION THREAD

**Discussion by corpix** [gravitational/teleport#10391] [Q&A] [answered]
Hello. I have a problem using teleport behind L7 load balancer.
I have successfully run web UI behind reverse proxy with new TLS-routing feature turned on.
But applications is not working.

I get redirect to port 3080: https://teleport.corpix.dev:3080/web/launch/dumper.teleport.corpix.dev/...

Is there any configuration I could apply to make it work?
- maybe configure proxy service to use separate hostname (like teleport-proxy.corpix.dev)?
- OR play nicely with teleport TLS-routing?

I use reverse proxy for many reasons and I feel strange about teleport not supporting such a common thing. Also I can't find any documentation on this topic.

**Comment by corpix** [comment]
I have resolved part of the problem:
```yaml
proxy_service:
  acme:
    email: root@corpix.dev
    enabled: 'no'
  enabled: 'yes'
  https_keypairs: []
  public_addr: teleport.corpix.dev:443
  web_listen_addr: :443
```

Port number in `public_addr` is mandatory. Otherwise we will see redirect to port 3080.

But now comes the interesting part. As I see teleport will use this `public_addr` to create reverse tunnels which means it will connect through internet and my L7 balancer.
This is strange.

Any way to specify non-public hostname to create reverse tunnels?

**UPD:** if there is a way to connect to internal hostname this probably will not work with `acme.enabled = no` because... this is certificate which is used by teleport (pay attention to `CN` and `Alternative Name`):

```
root@bastion-0 ~# openssl x509 -noout -text -in /mnt/data/teleport/webproxy_cert.pem
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            ...
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = US, O = localhost, CN = localhost
        Validity
            Not Before: Feb 16 19:19:01 2022 GMT
            Not After : Feb 14 19:19:01 2032 GMT
        Subject: C = US, O = localhost, CN = localhost
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)
                Modulus:
                    ...
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment, Certificate Sign
            X509v3 Basic Constraints: critical
                CA:TRUE
            X509v3 Subject Key Identifier:
                ...
            X509v3 Subject Alternative Name:
                DNS:bastion-0, DNS:localhost, DNS:localhost.local, IP Address:0:0:0:0:0:0:0:1, IP Address:127.0.0.1, IP Address:0:0:0:0:0:0:0:1
```

**Comment by webvictim** [comment, marked as answer]
You can set `proxy_service.tunnel_public_addr` to define the address Teleport will connect to in order to create reverse tunnels.
